Automate RVA ID Validation

mzack5020 opened the pull request:
<!--- Provide a general summary of your changes in the Title above -->

## 🗣 Description
Removed dependency on the user providing a file with the proper format. Previously, the user must provide a file that began with the correctly formatted RVA ID. The RVA ID validation now uses a regular expression to validate the RVA ID within the data itself. 

This pull request resolves issue #12.

## 💭 Motivation and Context
#12 Requiring the user to provide a properly formatted filename is an unnecessary manual step that adds additional steps.

## 🧪 Testing
Tested the new RVA ID validation logic using my local Python environment. Additionally, @mcdonnnj tested using a complete deploy using the Terraform code for this project. 

## 📷 Screenshots (if appropriate)

## 🚥 Types of Changes
- [X] Breaking change (causes existing functionality to change)

## ✅ Checklist
- [X] My code follows the code style of this project.
- [X] My change requires a change to the documentation.
- [X] I have updated the documentation accordingly.
- [X] I have read the **CONTRIBUTING** document.
- [X] I have added tests to cover my changes.
- [X] All new and existing tests passed.

mzack5020 committed:
Pushed in order to share current progress with team.

mzack5020 committed:
Merge branch 'develop' of github.com:cisagov/findings-data-import-lambda into AutomateRvValidation

mzack5020 committed:
Testing signed commit by making minor change.

mzack5020 committed:
Testing signed commit by making minor change.

mzack5020 committed:
Testing signed commit by making minor change.

mzack5020 committed:
Added in RV number validation and fixed regex problems.

mzack5020 committed:
Reduced RVA Number validator to enhance performance and fit requirements.

mzack5020 committed:
Removed Exception in lieu of logging error and skipping record.

mzack5020 committed:
Added to the README to include the RVA ID calculation so the user understands how the RVA ID is inserted into the database.

mcdonnnj committed:
Adjust regex expression to account for another ID format.
Remove unneeded variable and switch to a continue statement.

mcdonnnj committed:
Adjust phrasing of error message when no ID is found in a record.

mcdonnnj committed:
Adjusted README descroption of ID handling.

jsf9k commented on fdi/findings_data_import.py:
```diff
@@ -133,11 +134,6 @@ def import_data(
     temp_file_descriptor, temp_data_filepath = tempfile.mkstemp()
 
     try:
```
A potential improvement is to wrap each piece of code that could throw an exception in its own `try` block, and catch only the exceptions that can be thrown in the corresponding `catch` block.  That way you can give a context-specific error or warning message when you catch the exception, and be more specific about which exceptions you are catching.

This doesn't need to hold up this PR, but it's an idea for a future improvement.

dav3r replied:
@mzack5020 or @mcdonnnj - Can one of you please add a brief issue to this repo with @jsf9k's suggested improvement?

mcdonnnj replied:
Please see #15 for future work.

jsf9k reviewed (approved):
Approved, but please see my comment regarding a suggested future improvement.

felddy commented on fdi/findings_data_import.py:
```diff
@@ -199,10 +195,17 @@ def import_data(
                     finding.pop(field, None)
 
+            # Get RVA ID from last 4 digits
+            rvaId = re.search(r"(\d{4})(?:\.\d{1,})?$", finding["RVA ID"])
```
_Code never lies, comments sometimes do._
–Ron Jeffries

This comment does not match the function of the regular expression that follows.
See: https://regexr.com/4p7pc

Please rewrite to include the information in the `README.md`

dav3r reviewed (approved):
Looks good gentlemen! 👍

mcdonnnj committed:
Adjust comment to accurately describe what the regex is looking for to get an RVA ID.
Adjust regex to use \d+ instead of \d{1,} as they are functionally the same, but the former is clearer.
